source/globalCommands.py

```diff
@@ -3284,6 +3286,38 @@ def script_braille_cycleShowMessages(self, gesture: inputCore.InputGesture) -> N
 		msg = _("Braille show messages %s") % ShowMessages(newValue).displayString
 		ui.message(msg)
 
+	@script(
+		# Translators: Input help mode message for cycle through braille show selection command.
+		description=_("Cycle through the braille show selection states"),
+		category=SCRCAT_BRAILLE
+	)
+	def script_braille_cycleShowSelection(self, gesture: inputCore.InputGesture) -> None:
+		"""Set next state of braille show selection and reports it with ui.message."""
+		featureFlag: FeatureFlag = config.conf["braille"]["showSelection"]
+		boolFlag: BoolFlag = featureFlag.enumClassType
+		values = [x.value for x in boolFlag]
+		currentValue = featureFlag.value.value
+		nextValueIndex = (currentValue % len(values)) + 1
+		nextName: str = boolFlag(nextValueIndex).name
+		config.conf["braille"]["showSelection"] = nextName
+		featureFlag = config.conf["braille"]["showSelection"]
+		if featureFlag.isDefault():
+			displayString = featureFlag.behaviorOfDefault.displayString
+			# Translators: Used when reporting braille show selection feature flag
+			# default behavior.
+			msg = _("Braille show selection default (%s)") % featureFlag.behaviorOfDefault.displayString
+		else:
+			# Translators: Reports which show braille selection state is used
+			# (disabled or enabled).
+			msg = _("Braille show selection %s") % BoolFlag[nextName].displayString
+		ui.message(msg)
+
+	def _featureFlagDefaultBehaviorDisplayString(self, displayString: str) -> str:

```

Comment:
Purpose is to avoid situation where translators should translate same phrase ("Default") many times. As far as I understand, it would happen if there were more input gestures handling feature flag settings. I suppose that basic idea in nvdaControls.py is somewhat similar but implementation is different (in settings dialog default behavior alternative in combo box includes "Default"). I suppose that "Default" (or its translation) should be added automatically when default behavior state is selected. Compare with BoolFlag class function _displayStringLabels in featureFlagEnums.py

If it is better to remove _featureFlagDefaultBehaviorDisplayString function, no problem.

Reply:
yes, please remove it in favour of what was proposed here https://github.com/nvaccess/nvda/pull/14959#discussion_r1217463418

Reply:
As mentioned there, it is important that translators can re-arrange the full string, rather than in parts